engine/enginerpc/engine.proto

```diff
@@ -398,6 +398,113 @@ message EngineService {
+  bool notify_on_critical = 21 ;
+  bool notify_on_ok = 22 ;
+  bool notify_on_flappingstart = 23 ;
+  bool notify_on_flappingstop = 24 ;
+  bool notify_on_flappingdisabled = 25 ;
+  bool notify_on_downtime = 26 ;
+  bool stalk_on_ok = 27 ;
+  bool stalk_on_unknown = 28 ;
+  bool stalk_on_warning = 29 ;
+  bool stalk_on_critical = 30 ;
+  bool is_volatile = 31 ;
+  string notification_period = 32 ;
+  bool flap_detection_enabled = 33;
+  double low_flap_threshold = 34;
+  double high_flap_threshold = 35;
+  bool flap_detection_on_ok = 36;
+  bool flap_detection_on_warning = 37;
+  bool flap_detection_on_unknown = 38;
+  bool flap_detection_on_critical = 39;
+  int32 process_performance_data = 40;
+  bool check_freshness_enabled = 41;
+  int32 freshness_threshold = 42;
+  bool passive_checks_enabled = 43;
+  bool event_handler_enabled = 44;
+  bool active_checks_enabled = 45;
+  int32 retain_status_information = 46;
+  bool retain_nonstatus_information = 47;
+  bool notifications_enabled = 48;
+  bool obsess_over = 49;
+  string notes = 50;
+  string notes_url = 51;
+  string action_url = 52;
+  string icon_image = 53;
+  string icon_image_alt = 54;
+  bool problem_has_been_acknowledged = 55;
+  enum AckType { 
+    NONE = 0; 
+    NORMAL = 1; 
+    STICKY = 2;
+  }
+  AckType acknowledgement = 56;
+  bool host_problem_at_last_check = 57;
+  enum CheckType {
+    CHECK_ACTIVE = 0;
+    CHECK_PASSIVE = 1; 
+  }
+  CheckType check_type = 58;
+  State last_state = 59;
+  State last_hard_state = 60;
+  string plugin_output = 61;
+  string long_plugin_output = 62;
+  string perf_data = 63;
+  State state_type = 64;
+  string next_check = 66;
+  bool should_be_scheduled = 67;
+  string last_check = 68;
+  int32 current_attempt = 69;
+  uint64 current_event_id = 70;
+  uint64 last_event_id = 71;
+  uint64 current_problem_id = 72;
+  uint64 last_problem_id = 73;
+  string last_notification = 74;
+  string next_notification = 75;
+  bool  no_more_notifications = 76;
+  string last_state_change = 77;
+  string last_hard_state_change = 78;
+  string last_time_ok = 79;
+  string last_time_warning = 80;
+  string last_time_unknown = 81;
+  string last_time_critical = 82;
+  bool has_been_checked = 83;
+  bool is_being_freshened = 84;
+  bool notified_on_unknown = 85;
+  bool notified_on_warning = 86;
+  bool notified_on_critical = 87;
+  int32 notification_number = 88;
+  uint64 current_notification_id = 89;
+  double latency = 90;
+  double execution_time = 91;
+  bool is_executing = 92;
+  int32 check_options = 93;
+  int32 scheduled_downtime_depth = 94;
+  int32 pending_flex_downtime = 95;
+  string state_history = 96;
+  uint32 state_history_index = 97;
+  bool is_flapping = 98;
+  uint64 flapping_comment_id = 99;
+  double percent_state_change = 100;
+  uint32 modified_attributes = 101;
+  string host_ptr = 102;
+  string event_handler_args = 103;
+  string check_command_args = 104;
+  repeated string custom_variables = 105;
```

Comment:
protobuf defines a map that you can use here.
something like
```
map<string,string> custom_variables = 105;
```
Then we could have a view like "key=value"